rest-service/manager_rest/resource_manager.py

```diff
@@ -670,41 +663,25 @@ def delete_deployment_environment(self, deployment_id,
                 if execution.status not in ExecutionState.END_STATES
             )
             raise manager_exceptions.DependentExistsError(
-                f"Can't delete deployment {deployment_id} - There are "
+                f"Can't delete deployment {deployment.id} - There are "
                 f"running or queued executions for this deployment. "
                 f"Running executions ids: {running_ids}"
             )
 
         if not force:
-            node_instances = self.sm.list(models.NodeInstance, filters={
-                'deployment_id': deployment_id
-            }, get_all_results=True)
             # validate either all nodes for this deployment are still
             # uninitialized or have been deleted
-            if any(node.state not in ('uninitialized', 'deleted') for node in
-                   node_instances):
-                existing_instances = ','.join(
-                    node.id for node in node_instances
-                    if node.state not in ('uninitialized', 'deleted')
-                )
+            node_instances = self.sm.list(models.NodeInstance, filters={
```

Comment:
I wonder if this should be something that can be retrieved from the deployment object itself?
deployment.node_instances?

Reply:
it would be good, but i dont know how to do it just yet. Remember that it's a dep->node->instance

additionally, this one also has the state filter on it